tuf/utils/x509.go

```diff
@@ -443,6 +454,9 @@ func EncryptPrivateKey(key data.PrivateKey, role, passphrase string) ([]byte, er
 		return nil, fmt.Errorf("unable to encrypt key - invalid PEM file produced")
 	}
 	encryptedPEMBlock.Headers["role"] = role
+	if gun != "" {
```

Comment:
@riyazdf I thought for delegations we set the role to `"user"`. If that's not the case then yeah, we should make the header not appear.